src/sentry/models/dsymfile.py

```diff
@@ -297,30 +297,34 @@ def create_dsym_from_id(project, dsym_type, cpu_name, debug_id,
             if rv.file.checksum == checksum:
                 return rv, False
         except ProjectDSymFile.DoesNotExist:
-            pass
-        else:
-            # The checksum mismatches.  In this case we delete the old object
-            # and perform a re-upload.
-            rv.delete()
+            rv = None
 
         file = File.objects.create(
             name=debug_id,
             type='project.dsym',
             headers={'Content-Type': DSYM_MIMETYPES[dsym_type]},
         )
         file.putfile(fileobj)
-        try:
-            with transaction.atomic():
-                rv = ProjectDSymFile.objects.create(
-                    file=file,
-                    debug_id=debug_id,
-                    cpu_name=cpu_name,
-                    object_name=object_name,
-                    project=project,
-                )
-        except IntegrityError:
-            file.delete()
-            rv = ProjectDSymFile.objects.get(debug_id=debug_id, project=project)
+
+        kwargs = {
+            'file': file,
+            'debug_id': debug_id,
+            'cpu_name': cpu_name,
+            'object_name': object_name,
+            'project': project
+        }
+
+        if rv is None:
+            try:
+                with transaction.atomic():
+                    rv = ProjectDSymFile.objects.create(**kwargs)
+            except IntegrityError:
+                file.delete()
+                rv = ProjectDSymFile.objects.get(debug_id=debug_id, project=project)
```

Comment:
This means we keep the old file now rather than the new one.

Reply:
It‘s the same as before.

Reply:
How so? Before we deleted the old file, now we delete the new one no?

Reply:
I might be missing something obvious here, but this is the relevant section in the old code: https://github.com/getsentry/sentry/blob/93b0e5b4332c241f25af4c63062090001c35f08e/src/sentry/models/dsymfile.py#L322-L323

This branch handles if two concurrent jobs try to add a file at the same time and just lets the faster job win.